src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -416,7 +428,54 @@ private void RenderUnboundAttribute(string attributeName, Chunk attributeValueCh
                 // C# code, then we need to buffer it.
                 if (!isPlainTextValue)
                 {
-                    BuildBufferedWritingScope(attributeValueChunk, htmlEncodeValues: true);
+                    BuildBufferedWritingScope(renderValue: () =>
+                    {
+                        IChunkVisitor visitor;
+                        var parentChunk = attributeValueChunk as ParentChunk;
+                        containsDynamicCodeAttributeChunk = parentChunk
+                            .Children
+                            .Any(child => child is DynamicCodeAttributeChunk);
+
+                        if (!_designTimeMode && containsDynamicCodeAttributeChunk)
+                        {
+                            // If there are any non-dynamic code attribute chunks at the top level of the attribute
+                            // value's chunk it means there's static content in the value, the attribute will always
+                            // render.
```

Comment:
Comment describes the opposite case of this block. Reword to cover what's important when `containsDynamicCodeAttributeChunk` is `true`.